tsfc/kernel_interface/firedrake_loopy.py

```diff
@@ -358,50 +420,80 @@ def construct_kernel(self, name, ctx, log=False):
         # Add return arg
         funarg = self.generate_arg_from_expression(self.return_variables)
         args = [kernel_args.OutputKernelArg(funarg)]
-        # Add coordinates arg
-        coord = self.domain_coordinate[info.domain]
-        expr = self.coefficient_map[coord]
-        funarg = self.generate_arg_from_expression(expr)
-        args.append(kernel_args.CoordinatesKernelArg(funarg))
-        if oriented:
-            funarg = self.generate_arg_from_expression(self._cell_orientations, dtype=numpy.int32)
-            args.append(kernel_args.CellOrientationsKernelArg(funarg))
-        if needs_cell_sizes:
-            funarg = self.generate_arg_from_expression(self._cell_sizes)
-            args.append(kernel_args.CellSizesKernelArg(funarg))
+        active_domain_numbers_coordinates, args_ = self.make_active_domain_numbers({d: self.coefficient_map[c] for d, c in self.domain_coordinate.items()},
+                                                                                   active_variables,
+                                                                                   kernel_args.CoordinatesKernelArg)
+        args.extend(args_)
```

Comment:
Note: After calling `self.compile_gem()`, we use `self.make_active_domain_numbers()` to get indices of the active domains.